index.js

```diff
@@ -1,255 +1,48 @@
-    if (!config.key) return callback('Missing LogDNA Ingestion Key');
-
-    // Prepare HTTP Request Options
-    const options = {
-        url: LOGDNA_URL
-        , qs: config.tags ? {
-            tags: config.tags
-            , hostname: config.hostname
-        } : {
-            hostname: config.hostname
-        }, method: 'POST'
-        , body: JSON.stringify({
-            e: 'ls'
-            , ls: payload
-        }), auth: {
-            username: config.key
-        }, headers: {
-            'Content-Type': 'application/json; charset=UTF-8'
-            , 'user-agent': config.UserAgent
-        }, timeout: MAX_REQUEST_TIMEOUT_MS
-        , withCredentials: false
-        , agent: new agent.HttpsAgent({
-            freeSocketTimeout: FREE_SOCKET_TIMEOUT_MS
-        })
-    };
-
-    // Flush the Log
-    async.retry({
-        times: MAX_REQUEST_RETRIES
-        , interval: retryCount => REQUEST_RETRY_INTERVAL_MS * Math.pow(2, retryCount)
-        , errorFilter: errCode => DEFAULT_HTTP_ERRORS.includes(errCode) || errCode === 'INTERNAL_SERVER_ERROR'
-    }, (reqCallback) => {
-        return request(options, (error, response, body) => {
-            if (error) return reqCallback(error.code);
-            if (response.statusCode >= INTERNAL_SERVER_ERROR) return reqCallback('INTERNAL_SERVER_ERROR');
-            return reqCallback(null, body);
-        });
-    }, callback);
-};
-
-// Main Handler
-exports.handler = (event, context, callback) => {
-    const config = getConfig(event)
-        , eventData = processEvent(parseEvent(event))
-        , payload = config.eventlog ? [prepareEvent(eventData)] : []
-        , s3params = {
-            Bucket: eventData && eventData.meta && eventData.meta.bucket && eventData.meta.bucket.name || undefined
-            , Key: eventData && eventData.meta && eventData.meta.object && eventData.meta.object.key || undefined
-        };
-
-    if (config.filelog) {
-        return getLogs(s3params, (error, lines) => {
-            if (error) {
-                if (config.eventlog) return send(payload, config, callback);
-                return callback(error);
-            }
-
-            return async.everySeries(batchify(payload.concat(prepareLogs(lines, eventData))), (batch, next) => {
-                setTimeout(() => {
-                    return send(batch, config, next);
-                }, BATCH_INTERVAL_MS);
-            }, callback);
-        });
-    } else if (!config.eventlog) {
-        return callback('None of file and event logging has been enabled!');
-    }
 
-    return send(payload, config, callback);
-};
+        return callback(error)
+      }
+
+      const logArrays = payload.concat(transformer.prepareLogs(lines, eventData))
+      const batches = transformer.batchify(logArrays)
+      return async.everySeries(batches, (batch, next) => {
+        setTimeout(() => {
+          return logger.flush(batch, config, next)
+        }, BATCH_INTERVAL)
+      }, callback)
```

Comment:
@mdeltito, I remember I needed to add it since our ingester wasn't stable for sequential ingestions without any interval but I will replace the whole `logger` with our `logger-node`

Reply:
yeah I think that will help a lot, it already handles buffered flushing (default is 250ms I believe), retries, all that other fun stuff. the only difference being that it's event-based instead of callback based, but that shouldn't change much about this block.